pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -541,29 +542,49 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, StablePoolStorage,
         uint256,
         uint256[] memory scalingFactors,
         bytes memory userData
-    ) internal override returns (uint256, uint256[] memory) {
+    ) internal override returns (uint256 bptAmountOut, uint256[] memory amountsIn) {
         StablePhantomPoolUserData.JoinKindPhantom kind = userData.joinKind();
 
         (uint256 virtualSupply, uint256[] memory balancesWithoutBpt) = _payProtocolFeesBeforeJoinExit(balances);
+        (uint256 currentAmp, ) = _getAmplificationParameter();
 
+        // The individual join functions mutate balancesWithoutBpt so that they reflect the post-join balances
```

Comment:
Isn't this wrong? We do this ourselves later

Reply:
Ah - forgot to remove that comment (I did fix the comments below)